File: home_assignments/csharp/TicketSeller/model/validators/interfaces/IFestivalValidator.cs
using System;
using model.entities;

namespace model.validators.interfaces;

public interface IFestivalValidator:IValidator<int,Festival>
{
    /// <summary>
    /// validates the id of a festival
    /// </summary>
    /// <param name="id">said id</param>
    /// <exception cref="ValidationException">if id is null</exception>
    void ValidateId(int id);
    
    /// <summary>
    /// validates a date
    /// </summary>
    /// <param name="date">said date</param>
    /// <exception cref="ValidationException">if date is null</exception>
    void ValidateDate(DateTime date);

}

File: home_assignments/java/TicketSeller/TicketSellerModel/src/main/java/ticketseller/model/validators/interfaces/IFestivalValidator.java
package ticketseller.model.validators.interfaces;


import ticketseller.model.entities.Festival;
import ticketseller.model.validators.ValidationException;

import java.time.LocalDate;

public interface IFestivalValidator extends IValidator<Integer, Festival> {
    /**
     * validates a festival's id
     *
     * @param id - said id
     * @throws ValidationException if festival's id is null
     */
    void validateID(Integer id) throws ValidationException;

    /**
     * validates a date
     *
     * @param date - said date
     * @throws ValidationException if date is null
     */
    void validateDate(LocalDate date) throws ValidationException;
}
